src/arg_parser.ts

```diff
@@ -62,43 +62,26 @@ export function extractArgumentsFromDenoArgs(
 export function extractOptionsFromDenoArgs(
   denoArgs: string[],
   optionsMap: Map<string, IOption>,
-): string[] {
+): string[] | undefined {
   const errors: string[] = [];
   const passedInOptions = getOptionsPassedIntoDenoArgs(denoArgs, optionsMap);
 
-  let isValue = false;
-  let lastOptionObject: IOption | null = null;
-  const optionsProcessed: string[] = [];
-
-  passedInOptions.forEach((option: string) => {
```

Comment:
@dXu23 , again sorry for being late. don't get much time to look at the lower priority modules. anyways, any reason you went with  different loops as opposed to keeping the .forEeach here? i ask because from a readability point of view, it's difficult for me to keep track of what's being iterated. im all for different loops, but reason i chose for each was so that i didn't have to add [i] to retrieve values. wondering if you found this approach to be better